go/extractor/util/extractor.go

```diff
@@ -0,0 +1,101 @@
+package util
+
+import (
+	"encoding/json"
+	"log"
+	"os"
+	"path/filepath"
+)
+
+// Gets the path of the JSON file in the database scratch directory that is used
+// to store extraction results.
+func extractionResultsPath() string {
+	return filepath.Join(ScratchDir(), "extraction.json")
```

Comment:
Suggestion: rather than have a single json file and a read-modify-write process, instead define a distinctive _directory_ name like `$SCRATCH/go-extraction-results` where the extractor creates a fresh-named file (use a tempfile library for this), then have the autobuilder or pre-finalize step read whatever files were created and amalgamate them. This would mean there are no concurrency concerns except creating a fresh filename, which the tempfile library already knows how to do.

Reply:
Note that this is already a suggestion in the "open questions" section in the PR description